bot/util/utilities.py

```diff
@@ -154,24 +154,145 @@ async def cembed(self, ctx, channel: str, *, json_string: str):
+        """Command handler for the `presence` subcommand `listening`.
+
+        This is a command that changes the bots Discord presence to a listening activity with the specified name. The
+        Discord status can also be set via the optional status argument.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context from which this command is invoked.
+            status (Optional[discord.Status]): The status which should be displayed.
+            activity_name (str): The name of the activity shown.
+        """
+        activity = discord.Activity(type=discord.ActivityType.listening, name=activity_name)
+        await self.bot.change_presence(activity=activity, status=status)
+
+    @change_discord_presence.command(name="playing")
+    async def change_discord_presence_playing(self, ctx: commands.Context,
+                                              status: Optional[discord.Status] = discord.Status.online,
+                                              *, activity_name: str):
+        """Command handler for the `presence` subcommand `playing`.
+
+        This is a command that changes the bots Discord presence to a playing activity with the specified name. The
+        Discord status can also be set via the optional status argument.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context from which this command is invoked.
+            status (Optional[discord.Status]): The status which should be displayed.
+            activity_name (str): The name of the activity shown.
+        """
+        activity = discord.Game(name=activity_name)
+        await self.bot.change_presence(activity=activity, status=status)
+
+    @change_discord_presence.command(name="streaming")
+    async def change_discord_presence_streaming(self, ctx: commands.Context, stream_url: str,
+                                                status: Optional[discord.Status] = discord.Status.online,
+                                                *, activity_name: str):
+        """Command handler for the `presence` subcommand `streaming`.
+
+        This is a command that changes the bots Discord presence to a streaming activity with the specified name and
+        stream URL. The Discord status can also be set via the optional status argument.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context from which this command is invoked.
+            stream_url (str): The URL of the stream. (The watch button will redirect to this link if clicked)
+            status (Optional[discord.Status]): The status which should be displayed.
+            activity_name (str): The name of the activity shown.
+        """
+        # Everything other than Twitch probably won't work because of a clientside bug in Discord.
+        # More info here: https://github.com/Rapptz/discord.py/issues/5118
+        activity = discord.Streaming(name=activity_name, url=stream_url)
+        if "twitch" in stream_url:
```

Comment:
Do we only support youtube and twicht? Are mixxer (facebook gaming) etc. not supported?

Reply:
YouTube and Twitch are the most common ones and it seems that as of right now Discord can't even handle anything correctly except Twitch, so why bother adding anything else.